simple_salesforce/api.py

```diff
@@ -423,6 +444,115 @@ def _call_salesforce(self, method, url, **kwargs):
 
         return result
 
+    def _call_bulk_api(self, method, url, **kwargs):
+        """Makes a call to the bulk api
+
+        Returns a `request.result` object.
+        """
+        result = self.request.request(
+            method, url, headers=self.bulk_headers, **kwargs)
+
+        if result.status_code >= 300:
+            _exception_handler(result)
+
+        return result
+
+    def _make_bulk_request(self, url, data):
+        """Wrapper function to call a bulk API.
+
+        Parses the response to a dict.
+
+        Returns a `request.result` and a parsed
+        `reponse_object` dict
+        """
+        response = self._call_bulk_api("POST", url, data=data)
+        response_object = xmltodict.parse(response.text)
+        return response, response_object
+
+    def _template_location(self, template_file_name):
+        """Location of the template file"""
+        return os.path.join(self.location, template_file_name)
+
+    def _close_job(self, job_id):
+        """Makes Salesforce call to close a job"""
+        with open(self._template_location('close_job.xml'),
+                  'r') as close_xml_file:
+            close_xml_string = close_xml_file.read()
+            url = self.bulk_url + "/" + job_id
```

Comment:
Nitpick: I prefer interpolation over addition for strings (i.e. `'foo %s' % bar`)